### PR TITLE
#8 Allow per-client SSL Configuration

### PR DESCRIPTION
**Pull request by anilgursel**
Thanks for your pull request.  Please review the following guidelines.

- [x] Title includes issue id.
- [x] Description of the change added.
- [x] Commits are squashed.
- [x] Tests added.
- [ ] Documentation added/updated.
- [x] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsSpec.scala [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+  val serverBinding = Await.result(Http().bindAndHandle(route, "localhost", 0,
+    ConnectionContext.https(sslContext("example.com.jks", "1234567890"))), awaitMax)
+  val port = serverBinding.localAddress.getPort
+
+  def sslContext(store: String, pw: String) = {
+    val password: Array[Char] = pw.toCharArray // do not store passwords in code, read them from somewhere safe!
+
+    val ks: KeyStore = KeyStore.getInstance("JKS")
```
Comment (akara) [inline squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsSpec.scala]: This line throws an exception. Just a note.

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    val keystore: InputStream = getClass.getClassLoader.getResourceAsStream("ClientFlowHttpsSpec/" + store)
+
+    require(keystore != null, "Keystore required!")
```
Comment (akara) [inline squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsSpec.scala]: We should be able to teach Codacy that a check for something not being null by itself is not a red-flag 😄

File: squbs-httpclient/src/main/scala/org/squbs/endpoint/Endpoint.scala [hunk 3]
```diff
@@ -28,16 +29,23 @@ import org.squbs.unicomplex.JMX
...
   /**
     * Java API
     */
   def create(s: String): Endpoint = Endpoint(new URI(s))
+
+  /**
+    * Java API
+    */
+  import scala.compat.java8.OptionConverters._
+  def create(s: String, sslContext: Optional[SSLContext]): Endpoint = Endpoint(new URI(s), sslContext.asScala)
```
Comment (akara) [inline squbs-httpclient/src/main/scala/org/squbs/endpoint/Endpoint.scala]: While I don't see any issue, I'm sure this code is not yet covered.

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsSpec.scala (hunk 1 above)
Comment (anilgursel) [inline squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsSpec.scala]: Given this is test code, it is ok.

File: squbs-httpclient/src/main/scala/org/squbs/endpoint/Endpoint.scala (hunk 3 above)
Comment (anilgursel) [inline squbs-httpclient/src/main/scala/org/squbs/endpoint/Endpoint.scala]: You're right.  For all the APIs, I need to add tests and make sure the usage for Java, I am hoping to get the coverage right at that time.

**Pull request merged**